From 7ff5d45a9b21ff2d9e36a40ee1bf866e1ec212ad Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Thu, 4 Apr 2019 10:03:29 -0700
Subject: [PATCH 0612/1851] uio: xilinx_ai_engine: Set the uio device as pdev
 driver data

commit 7645fcdcf6028f6d8b0102bd2e5792a3a1a8513d from
https://github.com/Xilinx/linux-xlnx.git

It's needed when the driver is removed.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Wendy Liang <wendy.liang@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/uio/uio_xilinx_ai_engine.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/uio/uio_xilinx_ai_engine.c b/drivers/uio/uio_xilinx_ai_engine.c
index 2a70b378b43b..69778d83c52f 100644
--- a/drivers/uio/uio_xilinx_ai_engine.c
+++ b/drivers/uio/uio_xilinx_ai_engine.c
@@ -133,6 +133,7 @@ static int xilinx_ai_engine_probe(struct platform_device *pdev)
 	if (ret)
 		goto err_out;
 	platform_set_drvdata(uio, pdata);
+	platform_set_drvdata(pdev, uio);
 
 	dev_info(&pdev->dev, "Xilinx AI Engine UIO driver probed");
 	return 0;
-- 
2.31.1

